internal/reconcilers/anchor.go

```diff
@@ -106,17 +107,11 @@ func (r *AnchorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	// If the subnamespace doesn't exist, create it.
-	if inst.Status.State == api.Missing {
-		if err := r.writeNamespace(ctx, log, nm, pnm); err != nil {
-			// Write the "Missing" state to the anchor status if the subnamespace
-			// cannot be created for some reason. Without it, the anchor status will
-			// remain empty by default.
-			if anchorErr := r.writeInstance(ctx, log, inst); anchorErr != nil {
-				log.Error(anchorErr, "while setting anchor state", "state", api.Missing, "reason", err)
-			}
-			return ctrl.Result{}, err
+	if err := r.updateNamespace(ctx, log, nm, pnm); err != nil {
```

Comment:
Note that this change only affects subnamespaces (e.g., child namespaces created via an Anchor), and not _all_ descendant namespaces (e.g., child namespaces that are created normally and then assigned a parent). See [here](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/docs/user-guide/concepts.md#basic-subns) for the difference.

As a result, this should probably go in `hierarchy_config.go`, not here.